plugins/common/tsm/tsm.go

```diff
@@ -0,0 +1,85 @@
+package tsm
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/apiclarity/apiclarity/plugins/common"
+	"github.com/apiclarity/trace-sampling-manager/api/client/client"
+	"github.com/apiclarity/trace-sampling-manager/api/client/client/operations"
+)
+
+type TraceSamplingManager struct {
+	TraceSamplingManagerClient *client.TraceSamplingManager
+	// local cache of hosts to trace
+	Hosts            []string
+	samplingInterval time.Duration
+	lock             sync.RWMutex
+}
+
+func Create(enableTLS bool, host string) (*TraceSamplingManager, error) {
+	var tlsOptions *common.ClientTLSOptions
+	if enableTLS {
+		tlsOptions = &common.ClientTLSOptions{
+			RootCAFileName: common.CACertFile,
+		}
+	}
+	apiClient, err := common.NewTraceSamplingAPIClient(host, tlsOptions)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new trace sampling api client: %v", err)
+	}
+
+	return &TraceSamplingManager{
+		TraceSamplingManagerClient: apiClient,
+		samplingInterval:           10 * time.Second,
+	}, nil
+}
+
+func (t *TraceSamplingManager) Start() {
+	go func() {
+		for {
+			select {
+			case <-time.After(t.samplingInterval):
+				params := operations.NewGetHostsToTraceParams()
+
+				response, err := t.TraceSamplingManagerClient.Operations.GetHostsToTrace(params)
+				if err != nil {
+					log.Errorf("Failed to get hosts to trace: %v", err)
+				} else {
+					t.setHosts(response.Payload.Hosts)
+				}
+			}
+		}
+	}()
+}
+
+func (t *TraceSamplingManager) ShouldTrace(host string) bool {
+	hosts := t.getHosts()
+	if len(hosts) == 0 {
+		return true
+	}
+
+	for _, s := range hosts {
+		if s == host {
```

Comment:
need to address wildcard tracing.
if  s == "*" ---> return true

Reply:
https://github.com/apiclarity/wasm-filters/pull/8/files#diff-76cb46a2e63642865c78be467b0fd441981e1c449b5e847160c3a166c908f120R562